File: VB/DXSample/Data/DataHelper.vb
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Windows.Media
Imports System.Xml.Serialization
Imports DevExpress.Utils
Imports DevExpress.Xpf.Core.Native
Imports DXSample.ViewModels

Namespace DXSample.Data


    <XmlRoot("Employees")> _
    Public Class DataHelper
        Inherits List(Of Employee)

        Private Shared ReadOnly EmployeeDepartmentImages As New List(Of String)() From {"administration", "inventory", "manufacturing", "quality", "research", "sales"}
        Private Shared imageFemaleSrc As ImageSource
        Private Shared imageMaleSrc As ImageSource

        Private Shared Serializer As New XmlSerializer(GetType(DataHelper))

        Public Shared Function GetDataStream() As Stream
            Return AssemblyHelper.GetResourceStream(GetType(DataHelper).Assembly, "Data/EmployeesWithPhoto.xml", True)
        End Function
        Public Shared Function GetEmployeeDepartmentImage(ByVal departmentName As String) As Object
            Dim imageName = EmployeeDepartmentImages.FirstOrDefault(Function(x) departmentName.ToLower().Contains(x))
            If String.IsNullOrEmpty(imageName) Then
                Return Nothing
            End If
            Return "/DXSample;component/Images/" & imageName & ".png"
        End Function

        Public Shared Function GetEmployees() As List(Of Employee)
            Return DirectCast(Serializer.Deserialize(GetDataStream()), List(Of Employee))
        End Function

        Public Shared ReadOnly Property ImageFemale() As ImageSource
            Get
                If imageFemaleSrc Is Nothing Then
                    imageFemaleSrc = ImageSourceHelper.GetImageSource(AssemblyHelper.GetResourceUri(GetType(DataHelper).Assembly, "Images/Person_Female.png"))
                End If
                Return imageFemaleSrc
            End Get
        End Property

        Public Shared ReadOnly Property ImageMale() As ImageSource
            Get
                If imageMaleSrc Is Nothing Then
                    imageMaleSrc = ImageSourceHelper.GetImageSource(AssemblyHelper.GetResourceUri(GetType(DataHelper).Assembly, "Images/Person_Male.png"))
                End If
                Return imageMaleSrc
            End Get
        End Property
    End Class
End Namespace

File: CS/DXSample/Data/DataHelper.cs
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Windows.Media;
using System.Xml.Serialization;
using DevExpress.Utils;
using DevExpress.Xpf.Core.Native;
using DXSample.ViewModels;

namespace DXSample.Data {


    [XmlRoot("Employees")]
    public class DataHelper : List<Employee> {

        static readonly List<string> EmployeeDepartmentImages = new List<string> { "administration", "inventory", "manufacturing", "quality", "research", "sales" };
        static ImageSource imageFemaleSrc;
        static ImageSource imageMaleSrc;

        static XmlSerializer Serializer = new XmlSerializer(typeof(DataHelper));

        public static Stream GetDataStream() { return AssemblyHelper.GetResourceStream(typeof(DataHelper).Assembly, "Data/EmployeesWithPhoto.xml", true); }
        public static object GetEmployeeDepartmentImage(string departmentName) {
            var imageName = EmployeeDepartmentImages
                .FirstOrDefault(x => departmentName.ToLower().Contains(x));
            if (string.IsNullOrEmpty(imageName))
                return null;
            return "/DXSample;component/Images/" + imageName + ".png";
        }

        public static List<Employee> GetEmployees() {
            return (List<Employee>)Serializer.Deserialize(GetDataStream());
        }

        public static ImageSource ImageFemale {
            get {
                if (imageFemaleSrc == null)
                    imageFemaleSrc = ImageSourceHelper.GetImageSource(AssemblyHelper.GetResourceUri(typeof(DataHelper).Assembly, "Images/Person_Female.png"));
                return imageFemaleSrc;
            }
        }

        public static ImageSource ImageMale {
            get {
                if (imageMaleSrc == null)
                    imageMaleSrc = ImageSourceHelper.GetImageSource(AssemblyHelper.GetResourceUri(typeof(DataHelper).Assembly, "Images/Person_Male.png"));
                return imageMaleSrc;
            }
        }
    }
}
